templates/kubernetes/terraform/modules/kubernetes/backend_service.tf

```diff
@@ -0,0 +1,24 @@
+data "aws_secretsmanager_secret" "by-name" {
+  name = "<% .Name %>_cf_keypair"
+}
+
+data "aws_secretsmanager_secret_version" "cf_keypair" {
+  secret_id = data.aws_secretsmanager_secret.<% .Name %>_cf_keypair.id
```

Comment:
You don't need the project name here, you are referring to the resource defined above. So it should be `data.aws_secretsmanager_secret.cf_keypair.id`